src/hooks/GoTProvider.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { fetchCharacters } from '../../services/GoT-API.js';
+
+const GoTContext = createContext();
+
+// wrapper to provide state to child components
+export const GoTProvider = ({ children }) => {
+  const [characters, setCharacters] = useState([]);
+  const [page, setPage] = useState(6);
+  const [fireIce, setFireIce] = useState('fire');
+
+  useEffect(() => {
+    fetchCharacters(page)
+      .then(results => setCharacters(results));
+  }, [page]);
+  
+  const updatePage = (by) => setPage(prevPage => prevPage + by);
+
+  const toggle = ({ target }) => {
+    if(target.checked) setFireIce('fire');
+    if(!target.checked) setFireIce('ice');
+
+  };
+
+  return (
+    <GoTContext.Provider value={{ characters, page, updatePage, toggle }}>
```

Comment:
You also need fireIce as one of the values in order for your custom hook below to work